packages/core/contracts-ovm/insured-bridge/implementation/BridgeAdmin.sol

```diff
@@ -191,9 +195,9 @@ contract BridgeAdmin is BridgeAdminInterface, Ownable, OVM_CrossDomainEnabled {
         require(_bridgePool != address(0), "BridgePool cannot be zero address");
         require(_getCollateralWhitelist().isOnWhitelist(address(_l1Token)), "Payment token not whitelisted");
 
-        L1TokenRelationships storage whitelistedToken = whitelistedTokens[_l1Token];
-        whitelistedToken.l2Token = _l2Token;
-        whitelistedToken.bridgePool = _bridgePool;
+        require(address(BridgePoolInterface(_bridgePool).l1Token()) == _l1Token, "Bridge pool has different L1 token");
```

Comment:
whitelisting now checks that the `bridgePool` provided has the correct `l1Token` set.